src/profile-logic/profile-data.js

```diff
@@ -360,23 +362,35 @@ export function getTimingsForPath(
     interval,
     isInvertedTree,
     thread,
+    unfilteredThread,
+    sampleIndexOffset,
     categories
   );
 }
 
 /**
  * This function returns the timings for a specific path. The algorithm is
  * adjusted when the call tree is inverted.
+ * Note that the unfilteredThread should be the original thread before any filtering
+ * (by range or other) happens. Also sampleIndexOffset needs to be properly
+ * specified and is the offset to be applied on thread's indexes to access
+ * the same samples in unfilteredThread.
  */
 export function getTimingsForCallNodeIndex(
   needleNodeIndex: IndexIntoCallNodeTable | null,
   { callNodeTable, stackIndexToCallNodeIndex }: CallNodeInfo,
   interval: Milliseconds,
   isInvertedTree: boolean,
   thread: Thread,
+  unfilteredThread: Thread,
+  sampleIndexOffset: number,
```

Comment:
Is it possible to use the range filtered thread instead? This would remove the need for the `sampleIndexOffset`.

Reply:
`getTimingsForCallNodeIndex` is called in 2 locations:
* in the flame graph (for tooltips) -> here we could use the range filtered thread indeed.
* in the sidebar -> the sidebar uses the data from the "preview filtered thread". There's no unfiltered version of this thread, because it derives of the filtered thread. That's why we need this offset.